lisa/features/security_profile.py

```diff
@@ -1,38 +1,78 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
-from dataclasses import dataclass
+from dataclasses import dataclass, field
+from enum import Enum
 from functools import partial
 from typing import Any, Type, cast
 
 from dataclasses_json import dataclass_json
 
-from lisa import schema
+from lisa import schema, search_space
 from lisa.feature import Feature
+from lisa.util import LisaException, constants, field_metadata
 
 FEATURE_NAME_SECURITY_PROFILE = "Security_Profile"
 
 
+class SecurityProfileType(str, Enum):
+    Standard = constants.SECURITY_PROFILE_NONE
+    CVM = constants.SECURITY_PROFILE_CVM
+    Boot = constants.SECURITY_PROFILE_BOOT
+
+
 @dataclass_json()
 @dataclass()
 class SecurityProfileSettings(schema.FeatureSettings):
     type: str = FEATURE_NAME_SECURITY_PROFILE
-    secure_boot_enabled: bool = False
+    security_profile: search_space.SetSpace[SecurityProfileType] = field(
+        default_factory=partial(
+            search_space.SetSpace,
+            items=[
+                SecurityProfileType.Standard,
+                SecurityProfileType.Boot,
+                SecurityProfileType.CVM,
+            ],
+        ),
+        metadata=field_metadata(
+            decoder=partial(
+                search_space.decode_set_space_by_type, base_type=SecurityProfileType
+            )
+        ),
+    )
 
     def __hash__(self) -> int:
         return hash(self._get_key())
 
     def _get_key(self) -> str:
-        return f"{self.type}/{self.secure_boot_enabled}"
+        return f"{self.type}/{self.security_profile}"
 
     def _generate_min_capability(self, capability: Any) -> Any:
+        assert isinstance(
+            capability, SecurityProfileSettings
+        ), f"actual: {type(capability)}"
+
+        self.security_profile.intersection_update(capability.security_profile)
         return self
 
 
 class SecurityProfile(Feature):
     @classmethod
     def on_before_deployment(cls, *args: Any, **kwargs: Any) -> None:
         settings = cast(SecurityProfileSettings, kwargs.get("settings"))
-        if settings.secure_boot_enabled:
+        runbook: Any = cast(Any, kwargs.get("arm_parameters")).nodes[0]
+        try:
```

Comment:
Sorry, I just found the previous bug. The `on_before_deployment` should be in the Azure feature level, not general here.

Reply:
What does this change?

Reply:
The lisa/features is platform independent. The arm template is azure only. So the change should be in Azure folder.